dbt_semantic_interfaces/implementations/semantic_model.py

```diff
@@ -169,15 +173,35 @@ def get_entity(self, entity_reference: LinkableElementReference) -> PydanticEnti
 
         raise ValueError(f"No entity with name ({entity_reference}) in semantic_model with name ({self.name})")
 
-    def checked_agg_time_dimension_for_measure(self, measure_reference: MeasureReference):  # noqa: D
+    def checked_agg_time_dimension_for_measure(  # noqa: D
+        self, measure_reference: MeasureReference
+    ) -> TimeDimensionReference:
         measure = self.get_measure(measure_reference=measure_reference)
-        if self.defaults is not None:
-            default_agg_time_dimesion = self.defaults.agg_time_dimension
+        default_agg_time_dimension = self.defaults.agg_time_dimension if self.defaults is not None else None
 
-        agg_time_dimension_name = measure.agg_time_dimension or default_agg_time_dimesion
+        agg_time_dimension_name = measure.agg_time_dimension or default_agg_time_dimension
         assert agg_time_dimension_name is not None, (
             f"Aggregation time dimension for measure {measure.name} is not set! This should either be set directly on "
             f"the measure specification in the model, or else defaulted to the primary time dimension in the data "
             f"source containing the measure."
         )
         return TimeDimensionReference(element_name=agg_time_dimension_name)
+
+    @property
+    def primary_entity_reference(self) -> Optional[EntityReference]:  # noqa: D
+        if self.primary_entity is not None:
+            return EntityReference(element_name=self.primary_entity)
+
+        for entity in self.entities:
+            if entity.type is EntityType.PRIMARY:
+                return entity.reference
+            elif (
+                entity.type is EntityType.UNIQUE
+                or entity.type is EntityType.FOREIGN
+                or entity.type is EntityType.NATURAL
+            ):
+                pass
+            else:
+                assert_values_exhausted(entity.type)
```

Comment:
Is the `assert_values_exhausted` workflow actually necessary here? I think it's safe to assumethat the only entity type that this should ever return a reference for would be `EntityType.PRIMARY`, right? Like we don't think we're gonna add another entity type that should count as primary. If that is a safe assumption shouldn't we just do
```
if entity.type is EntityType.PRIMARY:
  return entity.reference
else:
  pass
```

Reply:
This is the general pattern that we have been going with when switching on any enum to ensure type safety when an enum is changed. I'm sure there have been cases where we thought things would never change, but turned out to.